### ISSUE TITLE
Next 24 hours

### ISSUE THREAD

**Issue by nitorrep** [websylv/homeassistant-meteoswiss#5]
Hi !

Thank you for your work, it is very appreciated !

Do you think it would be possible to have an 24 (or 48) hours forecast hour by hour with temperature, rain, wind and "icone"?

Best !

**Comment by websylv** [comment]
i will have a look if it's not too complicate to get data from meteoswiss website
the issue is more to display the rain/wind in homeassistant the weather component  don't have data for rain/wind in forecast, and i'm not sure to have the time to write a full component.
Without this information it's enough for you ?

The idea for you is to replace dark-sky ?

**Comment by chris669** [comment]
WOW, I asked api to Meteo Swiss 2 years ago, answered their on line survey, they indicated availability in 2 more years. And today I find this, thanks!
Sure it would be interesting having also hours forecasts, and storms, alerts predictions. I use it to manage my sun covers, so when too much wind, they goes up. So very important to me to know the frequency of actualisation.
Thanks! Chris
![SwissMeteo_HA](https://user-images.githubusercontent.com/15046400/79069586-a403ac00-7ccf-11ea-8190-4687ee38c73d.jpg)

**Comment by chris669** [comment]
But, is data reliable? Or dummy data's? Now it indicates Sunny, and a lot of wind, reality now is Cloudy and Zero wind...
![Screenshot_20200412-172157](https://user-images.githubusercontent.com/15046400/79072683-9d326480-7ce2-11ea-8170-6d88ece9b21d.png)

**Comment by CamFlyerCH** [comment]
> But, is data reliable? Or dummy data's? Now it indicates Sunny, and a lot of wind, reality now is Cloudy and Zero wind...

Compare it with the MeteoSwiss APP or Website. Probably you see the data of Geneva like described in Issue 4 https://github.com/websylv/homeassistant-meteoswiss/issues/4

**Comment by chris669** [comment]
I'm not sure, Cointrin was also cloudy around 17:30. Will try to force with Sion.
![Screenshot_20200412-183040](https://user-images.githubusercontent.com/15046400/79074325-066aa580-7cec-11ea-9843-c5233bb77238.png)

**Comment by chris669** [comment]
Ok, now without forcing, I confirm I see Sion, same datas, cannot explain what I had at 17:30.
![Screenshot_20200412-183621](https://user-images.githubusercontent.com/15046400/79074449-cb1ca680-7cec-11ea-85fd-22a8859050ab.png)
![Screenshot_20200412-183557](https://user-images.githubusercontent.com/15046400/79074463-da035900-7cec-11ea-8386-9edf73dcbfae.png)

**Comment by nitorrep** [comment]
> i will have a look if it's not too complicate to get data from meteoswiss website
> the issue is more to display the rain/wind in homeassistant the weather component don't have data for rain/wind in forecast, and i'm not sure to have the time to write a full component.
> Without this information it's enough for you ?
> 
> The idea for you is to replace dark-sky ?

Yeah somewhat if we could have something like dark-sky it would be awesome.

**Comment by chris669** [comment]
Hi, would it be also possible to get the temphigh of the **current day**? Now we have only the actual temp. Thanks.

**Comment by swissMack** [comment]
Hi, I am being dumb and a noob here, but how do I integrate this into Home Assistant?

**Comment by chris669** [comment]
copy the folder in custom_components and modify the configuration file.

**Comment by websylv** [comment]
or better than manual you can use HACS
https://hacs.xyz/docs/installation/manual

Like this you have updates when a new version is releasead

**Comment by colinfrei** [comment]
I just want to add my voice to request the expected max temperature of the current day.

**Comment by YvesZH** [comment]
Hello
Thanks also for your great work.  Much like @nitorrep wrote, I want a sensor with temperature and Wendspeed forecast for the next viewport.  I would also like a forecast today (so that the home assistant knows in the morning what the weather is like on the current day).  Thanks again!

**Comment by cyberatz** [comment]
Is there any update on this? Is there something I can help with?

**Comment by vistalba** [comment]
I'm looking for the same. As I want to use it for my suncovers I need temp-high/low for the current day. Just realized that there is just for the next day :( Any news on this?

**Comment by vistalba** [comment]
Basicly the data is available here for all stations:

```
https://data.geo.admin.ch/ch.meteoschweiz.messwerte-lufttemperatur-24h-min-1h/ch.meteoschweiz.messwerte-lufttemperatur-24h-min-1h_de.json

https://data.geo.admin.ch/ch.meteoschweiz.messwerte-lufttemperatur-24h-max-1h/ch.meteoschweiz.messwerte-lufttemperatur-24h-max-1h_de.json
``` 

It is not a very nice solution but you could use RESTful sensor and add each one. Just parse the json like:

`jq .features[28].properties` while 28 is JUN for Jungfraujoch. So you have to check which ID is for your station.

Unfortunately you can not use another place to get predicted min/max values as the source does only provide it for each station.

So I take a look how this is done in MeteoSwiss Apps:

So there is just a request with PLZ and add two 0 (e.g. 8001 -> 800100) and you get json result which can be parsed by jq for example:

```
$ curl -s https://app-prod-ws.meteoswiss-app.ch/v1/plzDetail?plz=800100 | jq .forecast[0].temperatureMin
15

$ curl -s https://app-prod-ws.meteoswiss-app.ch/v1/plzDetail?plz=800100 | jq .forecast[0].temperatureMax
22
```

I'm not sure if it is allowed to integrate such API requests in this integration? I can't find anything about usage of the API.

Anyone knows how to integrate this in the existing plugin?
